src/test/java/bio/terra/service/snapshot/flight/SnapshotAuthzIamStepTest.java

```diff
@@ -104,4 +111,36 @@ void testDoAndUndoStepWithDUOS() throws InterruptedException {
     assertThat(undoResult.getStepStatus(), equalTo(StepStatus.STEP_RESULT_SUCCESS));
     verify(iamService).deleteSnapshotResource(TEST_USER, SNAPSHOT_ID);
   }
+
+  @Test
+  void testDoAndUndoWithSnapshotFirecloudGroup() throws InterruptedException {
+    workingMap.put(
+        SnapshotWorkingMapKeys.SNAPSHOT_FIRECLOUD_GROUP_EMAIL, SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
+    when(flightContext.getWorkingMap()).thenReturn(workingMap);
+    overrideSnapshotRequestMode(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID);
+    var expectedPolicies =
+        new SnapshotRequestModelPolicies().addReadersItem(SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
+    Map<IamRole, String> expectedPoliciesMap = new HashMap<>();
+    expectedPoliciesMap.put(IamRole.READER, SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
```

Comment:
Does using `Map.of()` not work here?
```suggestion
    var expectedPoliciesMap = Map.of(IamRole.READER, SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
```

Reply:
It does not - I run into a deserialization error if I use Map.of().